scripts/generate-user-email-list.py

```diff
@@ -1,27 +1,61 @@
 """
 
 Usage:
-    scripts/generate-user-email-list.py <data_api_url> <data_api_token>
+    scripts/generate-user-email-list.py <data_api_url> <data_api_token> [--framework=<slug>]
 """
 import csv
 import sys
+import multiprocessing
+from multiprocessing.pool import ThreadPool
 
 from docopt import docopt
 from dmutils.apiclient import DataAPIClient
+from dmutils.audit import AuditTypes
 
 
-def generate_user_email_list(data_api_url, data_api_token):
-    client = DataAPIClient(data_api_url, data_api_token)
+def is_registered_with_framework(client, framework_id):
+    def is_of_framework(audit):
+        return audit['data'].get('frameworkSlug') == framework_id
+
+    def is_registered(user):
+        audits = client.find_audit_events(
+            audit_type=AuditTypes.register_framework_interest,
+            object_type='suppliers',
+            object_id=user['supplier']['supplierId'])
+        return any(map(is_of_framework, audits['auditEvents'])), user
+    return is_registered
 
-    writer = csv.writer(sys.stdout, delimiter=',', quotechar='"')
 
+def filter_is_registered_with_framework(client, framework_id, users):
+    pool = ThreadPool(100)
```

Comment:
A 100 threads might be quite a bit of load to the production API, especially since we haven't really tested the audit-events performance. Is loading all `register_framework_interest` events first and then filtering suppliers using the in-memory collection more complicated?